homeassistant/components/overkiz/entity.py

```diff
@@ -103,6 +103,9 @@ def __init__(
         self.entity_description = description
         self._attr_unique_id = f"{super().unique_id}-{self.entity_description.key}"
 
+        if self.entity_description.name:
+            self._attr_name = self.entity_description.name
```

Comment:
That is the default behavior. If this is needed, something else is wrong.

See:

https://github.com/home-assistant/core/blob/e360a4fa9e40ace622e31f309af265fc4d6bbcfd/homeassistant/helpers/entity.py#L318-L325

Reply:
@tetienne I think the regression is for devices that have multiple subdevices (#1, #2, etc.). I don't have any of these devices, but I am pretty sure this is where the regression is.

Reply:
OK so that what @nyroDev suggested.

Reply:
@tetienne are you able to test this? Since you have a device that has multiple subdevices.

Reply:
I will have a deeper look indeed and compare with my production instance.

Reply:
@tetienne this change don't change anything for the problem I reported.  

The best I can think of is this change:

```python
# entity.py
        if "#" in self.device_url and not self.device_url.endswith("#1"):
            # Only return the url of the base device, to inherit device name and model from parent device.

            # Adding this line vvv
            self._attr_name = self.device.label

            return {
                "identifiers": {(DOMAIN, self.executor.base_device_url)},
            }
```

This seems to do the trick for me.

Doing that give back the correct name to my sub entities like before.  

The entities ID are prefixed with the master entity, for instance in my case: `climate.hp_actuator_plancher`  
I don't know if it's an issue regarding the new naming convention requirement.

Reply:
Doing that do what I except for me:
```python
# entity.py
        if "#" in self.device_url and not self.device_url.endswith("#1"):
            # Only return the url of the base device, to inherit device name and model from parent device.
            self._attr_name = self.device.label
            return {
                "name": "",
                "identifiers": {(DOMAIN, self.device_url)},
            }
```
![Capture d’écran de 2022-09-28 13-04-08](https://user-images.githubusercontent.com/265662/192763524-6bc64898-5293-4723-961d-4069644984a1.png)

Name almost like before : 
WAS : Temp RDC Temperature
NOW: Temp RDC
(I guess that's the main point of the original PR)

ID changed (but do we have to care ?) : 
WAS : sensor.temp_rdc_temperature
NOW : sensor.hp_actuator_temp_rdc
The prefix `hp_actuaor` might be a problem, I'm not sure.